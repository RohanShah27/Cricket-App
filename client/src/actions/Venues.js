import { GET_VENUE_BYCOUNTRY } from "./types";
import axios from "axios";
const url = "http://localhost:5000/api/venues";

export const getVenuesByCountry = countryName => dispatch => {
  console.log("action " + countryName);
  return axios.get(url + "/all", countryName).then(res => {
    dispatch({
      type: GET_VENUE_BYCOUNTRY,
      payload: res.data.data
    });
    console.log("data " + res.data.data);
  });
};
