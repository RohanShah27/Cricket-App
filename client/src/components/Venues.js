// import React, { Component } from "react";
// import { connect } from "react-redux";
// import "../styles/Venues.css";
// import IndiaFlag from "../assests/india.jpg";
// import PakistanFlag from "../assests/pakistan.jpg";
// import SouthAfricaFlag from "../assests/southafrica.jpg";
// import SriLankaFlag from "../assests/srilanka.jpg";
// import AustraliaFlag from "../assests/australia.jpg";
// import BangladeshFlag from "../assests/bangladesh.png";
// import WestIndiesFlag from "../assests/westindies.png";
// import ZimbabweFlag from "../assests/zimbabwe.png";
// import NewZealandFlag from "../assests/newzealand.jpg";
// import EnglandFlag from "../assests/england.png";
// import Carousel from "react-multi-carousel";
// import CaptainRoop from "../assests/Captain Roop Singh.jpg";
// import Brabourne from "../assests/Brabourne Stadium.jpg";
// import Reliance from "../assests/Reliance stadium.jpg";
// import Nehru from "../assests/Nehru Stadium.jpg";
// import GreenPark from "../assests/green park.jpg";
// import "react-multi-carousel/lib/styles.css";
// import ReactCountryFlag from "react-country-flag";

// import { getVenuesByCountry } from "../actions/Venues";

// export class Venues extends Component {
//   responsive = {
//     desktop: {
//       breakpoint: { max: 3000, min: 1024 },
//       items: 8
//     }
//   };

//   state = {
//     country: ""
//   };

//   componentDidMount() {
//     this.props.getVenuesByCountry();
//   }

//   handleChange = name => {
//     console.log(name);
//     this.setState({ country: name });
//     let countryName = {
//       country: this.state.country
//     };
//     this.props.getVenuesByCountry(countryName);
//   };

//   render() {
//     // console.log(this.props.venues);
//     return (
//       <div>
//         <div>
//           <section className="venueSection">
//             <h1 className="venueHeading">Cricket Venues</h1>

//             <Carousel responsive={this.responsive}>
//               <ReactCountryFlag
//                 styleProps={{
//                   width: "80px",
//                   height: "80px"
//                 }}
//                 code="in"
//                 svg
//               />
//               {/* <div>
//                 <a
//                   value="India"
//                   onClick={() => {
//                     this.handleChange("India");
//                   }}
//                 >
//                   <img className="image" src={IndiaFlag} alt="Indian Flag " />
//                 </a>
//               </div>
//               <div>
//                 <img className="image" src={EnglandFlag} alt="Pakistan Flag " />
//               </div>
//               <div>
//                 <img
//                   className="image"
//                   src={WestIndiesFlag}
//                   alt="WestIndies Flag "
//                 />
//               </div>
//               <div>
//                 <img
//                   className="image"
//                   src={AustraliaFlag}
//                   alt="Australia Flag "
//                 />
//               </div>
//               <div>
//                 <img
//                   className="image"
//                   src={PakistanFlag}
//                   alt="Pakistan Flag "
//                 />
//               </div>
//               <div>
//                 <img
//                   className="image"
//                   src={BangladeshFlag}
//                   alt="Bangladesh Flag "
//                 />
//               </div>
//               <div>
//                 <img
//                   className="image"
//                   src={SriLankaFlag}
//                   alt="Sri Lanka Flag "
//                 />
//               </div>
//               <div>
//                 <img
//                   className="image"
//                   src={NewZealandFlag}
//                   alt="New Zealand Flag "
//                 />
//               </div>
//               <div>
//                 <img
//                   className="image"
//                   src={ZimbabweFlag}
//                   alt="Zimbabwe Flag "
//                 />
//               </div>
//               <div>
//                 <img
//                   className="image"
//                   src={SouthAfricaFlag}
//                   alt="South Africa Flag "
//                 />
//               </div> */}
//             </Carousel>
//           </section>
//         </div>
//         {/* <div>
//           <img className="image" src={IndiaFlag} alt="Indian Flag " />
//           <p className="countryHeading">Indian Venues</p>
//         </div> */}
//         <div class="grid-container">
//           <div class="groundCard">
//             <img className="groundImages" src={CaptainRoop} alt="Stadium " />
//             <div class="middle">
//               <div class="text">Gwalior</div>
//             </div>
//             <div class="container">
//               <h2 class="groundName">
//                 <b>Captain Roop Stadium</b>
//               </h2>
//             </div>
//           </div>
//           <div class="groundCard">
//             <img className="groundImages" src={Brabourne} alt="Stadium " />
//             <div class="container">
//               <h2 class="groundName">
//                 <b>Brabourne Stadium</b>
//               </h2>
//             </div>
//           </div>
//           <div class="groundCard">
//             <img className="groundImages" src={Reliance} alt="Stadium " />
//             <div class="container">
//               <h2 class="groundName">
//                 <b>Reliance Stadium</b>
//               </h2>
//             </div>
//           </div>
//           <div class="groundCard">
//             <img className="groundImages" src={GreenPark} alt="Stadium " />
//             <div class="container">
//               <h2 class="groundName">
//                 <b>Green Park Stadium</b>
//               </h2>
//             </div>
//           </div>
//           <div class="groundCard">
//             <img className="groundImages" src={Nehru} alt="Stadium " />
//             <div class="container">
//               <h2 class="groundName">
//                 <b>Nehru Stadium</b>
//               </h2>
//             </div>
//           </div>
//         </div>
//       </div>
//     );
//   }
// }

// const mapStateToProps = state => ({
//   venueByCountry: state.venuesReducer.venueByCountry
// });

// export default connect(
//   mapStateToProps,
//   { getVenuesByCountry }
// )(Venues);
