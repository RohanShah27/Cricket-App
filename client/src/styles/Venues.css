.venueSection {
  background-color: #484848;
  height: 255px;
  margin: 0px;
}
.venueHeading {
  font-family: sans-serif;
  color: #ffffff;
  padding-top: 30px;
  padding-left: 20px;
  font-size: 36px;
}
.venue-image {
  width: 100px;
  height: 100px;
  margin-left: 70px;
  margin-top: 25px;
  margin-bottom: 50px;
  border-radius: 50%;
}
.venue-image:hover {
  background: #ffffff;
  transform: scale(1.5);
}
.groundImages {
  margin-left: 60px;
  margin-top: 70px;
  /* border: 1px solid black; */
  width: 450px;
  height: 250px;
}
.groundCard {
  margin-left: 40px;

  width: 450px;
}
.middle {
  transition: 0.5s ease;
  opacity: 0;
  position: relative;
  /* left:20%; top:80%; */
  transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  text-align: center;
}
.groundCard:hover .groundImages {
  opacity: 0.3;
}
.groundCard:hover .middle {
  opacity: 1;
}
.grid-container {
  display: inline-grid;
  grid-template-columns: auto auto auto;
}
.countryHeading {
  float: right;
  margin-right: 74%;
  margin-top: 55px;
  font-size: 36px;
}
.groundName {
  margin-left: 170px;
}
