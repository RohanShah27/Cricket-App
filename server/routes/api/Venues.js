const express = require("express");
const router = express.Router();
const pg = require("pg-promise")();
const db = pg("postgre://postgres:123456@192.168.0.63:5432/crickstrait");

router.get("/all", async (req, res) => {
  const country = req.body.country;
  const result = await db.any(
    "select * from venues where country_name ='" + country + "';"
  );
  res.status(200).json({
    status: 200,
    data: result,
    message: "Retrieved all venues successfully"
  });
});

module.exports = router;
