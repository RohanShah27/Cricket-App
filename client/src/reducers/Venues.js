import { GET_VENUE_BYCOUNTRY } from "../actions/Types";

const initialState = {
  venueByCountry: []
};
export default function(state = initialState, action) {
  switch (action.type) {
    case GET_VENUE_BYCOUNTRY:
      return {
        ...state,
        venueByCountry: action.payload
      };
    default:
      return state;
  }
}
