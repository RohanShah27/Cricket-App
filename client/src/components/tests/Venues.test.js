import React from "react";
import { shallow } from "enzyme";
import { Venues } from "../Venues";

const Venue = [{}, {}];
const getVenuesByCountry = jest.fn();
const getVenues = jest.fn();
const wrapper = shallow(
  <Venues
    getVenues={getVenues}
    getVenuesByCountry={getVenuesByCountry}
    Venue={Venue}
  />
);

describe("Test Update user Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a section", () => {
    expect(wrapper.find("section").length).toBe(1);
  });

  it("should have exactly 11 counrties in the slider", () => {
    expect(wrapper.find(".image").length).toBe(11);
  });

  it("should have a Carousel", () => {
    expect(wrapper.find("Carousel").length).toBe(1);
  });

  it("contains h1", () => {
    expect(
      wrapper
        .find("h1")
        .at(0)
        .text()
    ).toEqual("Cricket Venues");
  });

  it("it should have button with text ", () => {
    expect(wrapper.find(".countryHeading").text()).toEqual("Indian Venues");
  });

  it("should have images in card", () => {
    expect(wrapper.find(".groundImages").length).toBe(5);
  });

  it("contains h2 tag which displays ground name", () => {
    expect(wrapper.find("h2").length).toBe(5);
  });

  it("stadium names should be in bold", () => {
    expect(wrapper.find("b").length).toBe(5);
  });

  it("it should have a picture of every ground/stadium", () => {
    expect(wrapper.find(".groundImages").length).toBe(5);
  });

  it("stadium should be displayed in a grid format", () => {
    expect(wrapper.find(".grid-container").length).toBe(0);
  });
});
