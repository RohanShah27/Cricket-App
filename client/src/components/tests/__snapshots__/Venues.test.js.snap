// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Update user Component should render the component 1`] = `ShallowWrapper {}`;
